### PR TITLE
[Merged by Bors] - Add linkcheck workflow

### LINKED ISSUE #2889: Broken links in book
## Description

Running [`linkcheck`](https://github.com/filiph/linkcheck) on the Lighthouse Book reveals a number of broken links:

```
$ linkcheck https://lighthouse-book.sigmaprime.io/
Done crawling.                   

https://lighthouse-book.sigmaprime.io/
- (151:42) 'local te..' => https://lighthouse-book.sigmaprime.io/local-testnets.html (HTTP 403)
- (151:42) 'local te..' => https://lighthouse-book.sigmaprime.io/local-testnets.html (HTTP 403)

https://lighthouse-book.sigmaprime.io/faq.html
- (191:14) '/beacon/..' => https://lighthouse-book.sigmaprime.io/http/beacon.html#beaconvalidatorsall (HTTP 403)
- (196:0) '/beacon/..' => https://lighthouse-book.sigmaprime.io/http/beacon.html#beaconvalidatorsall (HTTP 403)
- (260:54) 'here.' => https://lighthouse-book.sigmaprime.io/installation-source.html#updating-lighthouse (HTTP 200 but missing anchor)

https://lighthouse-book.sigmaprime.io/intro.html
- (151:42) 'local te..' => https://lighthouse-book.sigmaprime.io/local-testnets.html (HTTP 403)

https://lighthouse-book.sigmaprime.io/pi.html
- (154:15) 'Ubuntu D..' => https://lighthouse-book.sigmaprime.io/installation.html#dependencies-ubuntu (HTTP 200 but missing anchor)

https://lighthouse-book.sigmaprime.io/print.html
- (152:42) 'local te..' => https://lighthouse-book.sigmaprime.io/local-testnets.html (HTTP 403)
- (593:15) 'Ubuntu D..' => https://lighthouse-book.sigmaprime.io/installation.html#dependencies-ubuntu (HTTP 200 but missing anchor)
- (2145:7) 'Global V..' => https://lighthouse-book.sigmaprime.io/validator-inclusion.html#consensusglobal_votes (HTTP 200 but missing anchor)
- (3321:52) 'explicit..' => https://lighthouse-book.sigmaprime.io/subscribe-api (HTTP 403)
- (3686:14) '/beacon/..' => https://lighthouse-book.sigmaprime.io/http/beacon.html#beaconvalidatorsall (HTTP 403)
- (3691:0) '/beacon/..' => https://lighthouse-book.sigmaprime.io/http/beacon.html#beaconvalidatorsall (HTTP 403)
- (3755:54) 'here.' => https://lighthouse-book.sigmaprime.io/installation-source.html#updating-lighthouse (HTTP 200 but missing anchor)

https://lighthouse-book.sigmaprime.io/redundancy.html
- (211:52) 'explicit..' => https://lighthouse-book.sigmaprime.io/subscribe-api (HTTP 403)

https://lighthouse-book.sigmaprime.io/validator-inclusion.html
- (213:7) 'Global V..' => https://lighthouse-book.sigmaprime.io/validator-inclusion.html#consensusglobal_votes (HTTP 200 but missing anchor)

Errors. Checked 4448 links, 83 destination URLs (72 ignored), 3 have errors, 6 have warnings.
```

## Version

Book as of 1 January 2022

## Steps to resolve

Correct broken links by working out what they should point to instead. If feeling ambitious, try running `linkcheck` on a locally-hosted version of the book under CI.

### PR DESCRIPTION
**Pull request by ackintosh**
## Issue Addressed

Resolves #2889

## Additional Info

I have checked that the `linkcheck` workflow runs and detects broken links as expected, in https://github.com/ackintosh/lighthouse/pull/1.

### CODE REVIEW

**Review by michaelsproul — APPROVED** [review]
Awesome, thank you @ackintosh :blush:

File: .github/workflows/linkcheck.yml
Comment (michaelsproul) [inline .github/workflows/linkcheck.yml:9]: Do you mind adding a rule to run on `unstable`? It seems that we may as well run it slightly more often

Comment (ackintosh) [reply .github/workflows/linkcheck.yml:9]: It makes sense, thank you!

**Review by michaelsproul — APPROVED** [review]
Looks good!

**Comment by michaelsproul** [comment]
We'll merge this after v2.1.0 is released (there are lots of PRs blocked on this right now)

**Pull request closed without merging**
